.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+
+set -euo pipefail
+
+export CMAKE_QUIET="yes"
```

Comment:
Does this mean we can never see the compile commands?

That's a pretty big drawback compared to Jenkins - it after making some build update it won't be possible to use CI to see the effect on all platforms without having to find appropriate machines and manually build on all platforms.

The optimum here is probably to log the build commands to a file, then store that file as an artifact, so that the primary build output is not cluttered with compile commands but they're available when needed.

Reply:
Yes, it turns off cmake's verbose flag, resulting in a summary of the build commands e.g.
```
  | [  0%] Building CXX object lib/seccomp/CMakeFiles/MlSeccomp.dir/CSystemCallFilter_MacOSX.cc.o
  | [  0%] Building CXX object bin/controller/CMakeFiles/Mlcontroller.dir/CBlockingCallCancellingStreamMonitor.cc.o
  | [  0%] Building CXX object bin/categorize/CMakeFiles/Mlcategorize.dir/CCmdLineParser.cc.o
  | [  0%] Building CXX object lib/ver/CMakeFiles/MlVer.dir/CBuildInfo.cc.o
  | [  0%] Building CXX object bin/data_frame_analyzer/CMakeFiles/Mldata_frame_analyzer.dir/CCmdLineParser.cc.o
  | [  0%] Building CXX object bin/controller/CMakeFiles/Mlcontroller.dir/CCmdLineParser.cc.o
  | [  0%] Building CXX object bin/normalize/CMakeFiles/Mlnormalize.dir/CCmdLineParser.cc.o
  | [  1%] Building CXX object bin/controller/CMakeFiles/Mlcontroller.dir/CCommandProcessor.cc.o
  | [  1%] Building CXX object bin/controller/CMakeFiles/Mlcontroller.dir/CResponseJsonWriter.cc.o
  | [  1%] Building CXX object bin/autodetect/CMakeFiles/Mlautodetect.dir/CCmdLineParser.cc.o
  | [  1%] Building CXX object bin/pytorch_inference/CMakeFiles/Mlpytorch_inference.dir/CBufferedIStreamAdapter.cc.o
  | [  1%] Building CXX object bin/pytorch_inference/CMakeFiles/Mlpytorch_inference.dir/CCmdLineParser.cc.o
  | [  3%] Building CXX object bin/pytorch_inference/CMakeFiles/Mlpytorch_inference.dir/CCommandParser.cc.o
  | [  3%] Building CXX object bin/pytorch_inference/CMakeFiles/Mlpytorch_inference.dir/CThreadSettings.cc.o
  | [  3%] Building CXX object bin/pytorch_inference/CMakeFiles/Mlpytorch_inference.dir/CResultWriter.cc.o
```
With full build output BuildKite's console buffer fills up and you can no longer "follow" the output. However that's really just a niggle. I'll turn verbose output back on.

Reply:
> With full build output BuildKite's console buffer fills up and you can no longer "follow" the output.

That sounds like another bad problem.

Is there no way to log the full commands somewhere else but not log them to the console?

If there isn't then another alternative might be to have a build parameter so that usually the commands don't get logged but it's easy to kick off a job manually and set the parameter that switches on verbose output.

Reply:
Buildkite still retains the full log file and it is available for download, so there's no need to separately archive it.

As for optionally setting a build parameter controlling the flag, yes that is possible either via labels or comments. As it will require a little work on the buildkite PR hook and some additional code to parse the variables I propose to do this in a follow up PR...